android/src/com/google/zxing/client/android/wifi/WifiConfigManager.java

```diff
@@ -226,4 +251,30 @@ private static boolean isHexOfLength(CharSequence value, int... allowedLengths)
     return false;
   }
 
+  private static int parseEap(String eapString) {
+    if (eapString == null) {
+      return WifiEnterpriseConfig.Eap.NONE;
+    }
+    try {
+      return (Integer) WifiEnterpriseConfig.Eap.class.getField(eapString).get(null);
```

Comment:
Why was the reflection needed here? is it due to API levels?

Reply:
The advantage of reflection would be that zxing will support new methods without any changes. If you want, I can change this to an explicit list of methods without reflection, as all the important ones (PEAP, PWD, TLS, TTLS for Eap and MSCHAP, MSCHAPv2, PAP for Phase2) are available with API level 18.